torch/testing/_internal/common_utils.py

```diff
@@ -940,6 +940,35 @@ def get_comparison_dtype(a, b):
 
     return compare_dtype
 
+def compare_with_reference(torch_fn, ref_fn, sample_input):
+    # Compares torch function with reference function for given sample input (object of SampleInput)
+    # Note: only values are compares, type comparison is not done here
+    try:
+        n_inp, n_args, _ = sample_input.numpy()
+    except TypeError:
+        # TODO: This needs a comment
+        return
+    t_inp, t_args, _ = sample_input.splat()
+
+    t_kwargs, n_kwargs = torch_fn.sample_kwargs(t_inp.device, None, t_inp)
+    actual = torch_fn(t_inp, *t_args, **t_kwargs)
+    expected = ref_fn(n_inp, *n_args, **n_kwargs)
+
+    # Crafts a custom error message for smaller, printable tensors
+    if t_inp.numel() < 10:
+        msg = ("Failed to produce expected results! Input tensor was"
+               " {0}, torch result is {1}, and reference result is"
+               " {2}.").format(t_inp, actual, expected)
+    else:
+        msg = None
+
+    if isinstance(actual, torch.Tensor):
+        torch.testing.assert_close(actual, torch.as_tensor(expected, dtype=actual.dtype), msg=msg, check_device=False)
```

Comment:
Since the tensor coming from numpy will always be on the CPU, I suggest we simply also move the original tensor to the CPU instead of disabling the check.
```suggestion
        torch.testing.assert_close(actual.cpu(), torch.as_tensor(expected, dtype=actual.dtype), msg=msg)
```